### PR TITLE
kubeadm: Remove dependency on pkg/client/clientset_generated/internalclientset

### LINKED ISSUE #806: Drop dependency on `pkg/client/clientset_generated/internalclientset` from kubeadm
As part of the research into recursive dependencies of kubeadm in https://github.com/kubernetes/kubernetes/pull/63740/ we found that we depend on a lot of packages under `"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset`

Since we are using client-go, we need to find out why we depend on these packages and try to get rid of them

### PR DESCRIPTION
**Pull request by rosti**
**What this PR does / why we need it**:
This PR removes dependency in kubeadm on `pkg/client/clientset_generated/internalclientset` and blacklists it in `.import-restrictions`.
The above package is pulled in transitively by the usage of the following:

- [GetFlagString](https://github.com/kubernetes/kubernetes/blob/a481f4bbe828d7fdec95782c02df3743c1020820/cmd/kubeadm/app/cmd/version.go#L63) from `k8s.io/kubernetes/pkg/kubectl/cmd/util`

- [LeaseEndpointReconcilerType](https://github.com/kubernetes/kubernetes/blob/a481f4bbe828d7fdec95782c02df3743c1020820/cmd/kubeadm/app/phases/controlplane/manifests.go#L205) from `k8s.io/kubernetes/pkg/master/reconcilers`

Dropping the unneeded dependencies from the build shaves off 4-5MB from the amd64 executable and reduces the dependency hell in kubeadm.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes kubernetes/kubeadm#806

**Special notes for your reviewer**:
/cc @kubernetes/sig-cluster-lifecycle-pr-reviews
/area kubeadm
/assign @luxas
/assign @timothysc

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Review by luxas — APPROVED** [review]
Thanks _a lot_ @rosti! One super small nit only.
Can you report the before/after number for this command please :smile:?

```
go list -f '{{ join .Deps "\n" }}' "k8s.io/kubernetes/cmd/kubeadm" | wc -l
```
/approve

**Comment by rosti** [comment]
For `go list -f '{{ join .Deps "\n" }}' "k8s.io/kubernetes/cmd/kubeadm" | wc -l`

Before: 776
After: 618

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -255,6 +255,10 @@ const (
 	KubeAuditPolicyLogVolumeName = "audit-log"
 	// StaticPodAuditPolicyLogDir is the name of the directory in the static pod that will have the audit logs
 	StaticPodAuditPolicyLogDir = "/var/log/kubernetes/audit"
+
+	// LeaseEndpointReconcilerType will select a storage based reconciler
+	// Copied from pkg/master/reconcilers to avoid pulling extra dependencies
+	LeaseEndpointReconcilerType = "lease"
```
[the remark this answers is not in the document]
Comment (rosti) [reply cmd/kubeadm/app/constants/constants.go]: @luxas You mean to add TODO here, that mentions our preference to import this constant from a consts only package (without any transitive imports)?

**Review by luxas — APPROVED** [review]
/approve
/lgtm

The best feeling is when you either remove a large portion of code or a lot of dependencies :smile:. 
Thanks :+1:!!

**Pull request merged**